esp/bindings/http/platform/httpsecurecontext.cpp

```diff
@@ -0,0 +1,84 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "httpsecurecontext.hpp"
+
+const char* HTTP_PROPERTY_TYPE_COOKIE = "cookie";
```

Comment:
I thought about it, and even started refactoring for it, but ultimately decided against an enumeration. It can be changed if this is another way of saying "make it an enumeration," but here are my reasons for not doing it:

- A single enumeration defined at the interface level, and used as a parameter type, tends to make all implementations dependent on each other. My goal was to keep protocol-specific implementations as independent of one another as possible, so this was a non-starter.

- A protocol specific enumeration could be defined, but the top-level interface becomes 'int' instead of 'const char *'. That wouldn't be a problem, but I can envision developers taking shortcuts, not including the sub-class header file, and hard coding the type. I would much rather see 'getProp("cookie", ...)' hard-coded than 'getProp(0, ...)' in such cases.

- Given one, as yet unsupported, security manager that needs the function for two cookies, the overhead of a string comparison versus a switch statement seemed acceptable.